starter_code/src/movies.js

```diff
@@ -4,17 +4,68 @@
 
 // Get the average of all rates with 2 decimals 
 
+var totalRate = 0;
+
+function ratesAverage (movies) {
+
+var totalRate =  movies.reduce(function(accumulator, movie){
+      return accumulator + movie.rate;
+  }, 0); return  totalRate / movies.length; 
+}
 
 // Get the average of Drama Movies
 
+var dramaRate = movies.filter(function(movie){
+  if (movie.genre == 'Drama'){
+    return movie;
+  }
+});
+
+function dramaMoviesRate (dramaRate) {
+
+var totalRate =  dramaRate.reduce(function(accumulator, movie){
```

Comment:
Why not reuse the function above? Also, I would advise to contain the filter inside the function as well. You might receive an array different from the one that you have in the data file so you want to do all the steps inside a function.